crypto/evp/e_aes_cbc_hmac_sha1.c

```diff
@@ -859,6 +859,8 @@ static int aesni_cbc_hmac_sha1_ctrl(EVP_CIPHER_CTX *ctx, int type, int arg,
                 key->payload_length = len;
                 if ((key->aux.tls_ver =
                      p[arg - 4] << 8 | p[arg - 3]) >= TLS1_1_VERSION) {
+                    if (len < AES_BLOCK_SIZE)
+                        return 0;
```

Comment:
Good point.
In the code where return value of EVP_CIPHER_CTX_ctrl() is checked, 0 is treated as an error. -1 is treated as error only in few places.

In this case, the caller, tls1_enc() treats return value of <= 0 as an error. 

Man page do not have explicit mention about the return values of EVP_CIPHER_CTX_ctrl().